swarm_dial.go

```diff
@@ -312,6 +315,12 @@ func (s *Swarm) canDial(addr ma.Multiaddr) bool {
 
 // dial is the actual swarm's dial logic, gated by Dial.
 func (s *Swarm) dial(ctx context.Context, p peer.ID) (*Conn, error) {
+	// should we allow the connection to this peer at all ?
+	if s.ConnGater != nil && !s.ConnGater.InterceptPeerDial(p) {
+		log.Debugf("gater disallowed outbound connection to peer %s", p)
```

Comment:
Don't ask why...

```suggestion
		log.Debugf("gater disallowed outbound connection to peer %s", p.Pretty())
```

Reply:
@raulk Have moved this check to `Swarm.DialPeer` and have changed it to p.Pretty.